File: src/main/kotlin/chronicle/lastfm.kt
package chronicle

import kotlin.io

class LastFmImporter(username: String, apiKey: String) {
  companion object {
    // Name of the config file to specify Last.FM credentials
    val configFileName: String = "/lastfm.json"
  }

  fun import(): Unit {
    val config =
        javaClass
            .getResourceAsStream(configFileName)
            ?.reader(Charsets.UTF_8)
            ?.readText()
            ?: "Could not find $configFileName"
    println(config)
  }
}

File: src/main/scala/chronicle/lastfm.scala
package chronicle

import scala.io.Source

class LastFmImporter(username: String, apiKey: String) {
  final val configFileName: String = "/lastfm.json"
  
  def run(): Unit = {
    val config =
        Option(getClass.getResourceAsStream(configFileName))
            .map { stream => Source.fromInputStream(stream).mkString }
            .getOrElse(s"Could not read $configFileName")
    println(config)
  }
}
